### PR TITLE
Added NoCachingLookupFactory

### PR DESCRIPTION
**Pull request by jc89**
NoCachingLookupFactory: LookupFactory not using `Lookup` cache (Setting it to null)

### CODE REVIEW

File: src/main/java/com/spotify/dns/DnsSrvResolvers.java [hunk 1]
```diff
@@ -78,6 +82,10 @@ public DnsSrvResolver build() {
 
       LookupFactory lookupFactory = new SimpleLookupFactory();
 
+      if (!useLookupCache) {
```
Comment (pettermahlen) [inline src/main/java/com/spotify/dns/DnsSrvResolvers.java]: It looks like `useLookupCache`and `cacheLookups` conflict if both are set to true. Is that the case? The builder should disallow or handle that somehow, if so.

File: src/main/java/com/spotify/dns/NoCachingLookupFactory.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.spotify.dns;
+
+import com.google.common.base.Preconditions;
+import org.xbill.DNS.*;
```
Comment (pettermahlen) [inline src/main/java/com/spotify/dns/NoCachingLookupFactory.java]: Code style: don't do star imports.

File: src/test/java/com/spotify/dns/NoCachingLookupFactoryTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.xbill.DNS.*;
```
Comment (pettermahlen) [inline src/test/java/com/spotify/dns/NoCachingLookupFactoryTest.java]: Code style: don't do star imports.

File: src/test/java/com/spotify/dns/NoCachingLookupFactoryTest.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+import static org.hamcrest.CoreMatchers.*;
```
Comment (pettermahlen) [inline src/test/java/com/spotify/dns/NoCachingLookupFactoryTest.java]: Code style: don't do star imports.

File: src/test/java/com/spotify/dns/NoCachingLookupFactoryTest.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import static org.junit.Assert.assertThat;
+
+@RunWith(MockitoJUnitRunner.class)
+public class NoCachingLookupFactoryTest {
```
Comment (pettermahlen) [inline src/test/java/com/spotify/dns/NoCachingLookupFactoryTest.java]: This test class doesn't validate the key behaviour which I understand to be disabling the DNS cache. I expect that's pretty hard to test in practice. Did you try?

File: src/main/java/com/spotify/dns/DnsSrvResolvers.java (hunk 1 above)
Comment (pettermahlen) [inline src/main/java/com/spotify/dns/DnsSrvResolvers.java]: A test validating the conflict handling would also solve the code coverage regression.

Comment (jc89) [inline src/main/java/com/spotify/dns/DnsSrvResolvers.java]: Will do

File: src/test/java/com/spotify/dns/NoCachingLookupFactoryTest.java (hunk 5 above)
Comment (jc89) [inline src/test/java/com/spotify/dns/NoCachingLookupFactoryTest.java]: Im trying to find ways to automate this. A quick way i found to test this is basically asking for resolution of a name twice, and before the second run, disabling network in the workstation.

Using `SimpleLookupFactory` it does not fail, given that uses the Lookup cache.
Using `NoCachingLookupFactory` fails, given that is trying to resolve again.

**Comment by jc89** [comment]
@pettermahlen i think this could be useful when you are not interested in using the ttl declared in the records

I added a few test cases, will update the README later

**Comment by pettermahlen** [comment]
@jc89 looks good, the only thing left I think should be done is something like having the `DnsSrvResolverBuilder` throw an `IllegalStateException` if the user tries to set both `useLookupCache` and `cacheLookups` to true.

**Comment by jc89** [comment]
@pettermahlen by default, previously it was basically using the `Lookup` cache always. In order to keep that behaviour, i set the default value for that caching flag in the lookup to true.
If we add this validation, it could suddenly start breaking the build when people only update the version.
We could:
- Accept this fact and go ahead and add the validation. Would also need to update the usage examples
- Refactor the two flags into an enum, picking up the caching strategy to use (Like, ```cachingSchema(someSchema)``` where the schemas could be none, caching resolutions, caching lookups.

**Comment by pettermahlen** [comment]
Sorry, I was confused. The problematic case is `useLookupCache = false && cacheLookups = true`. In this case, the `useLookupCache` configuration will be silently ignored, which I think would be confusing to the user. 

Reading the above text, it's quite obvious that these two parameter names are very similar, and it's not clear what the difference between them really is. We should explain that somehow. Is it correct to say that:

- `useLookupCache` determines whether or not a given `Lookup` instance should cache its previous results, and
- `cacheLookups` determines whether a given thread should reuse `Lookup` instances for a given query between invocations.

**Pull request closed without merging**